Bring code up-to-date with the spec and fix a sorting bug

wessels opened the pull request.

wessels committed:
typo: original_digest[] -> original_digests[]

wessels committed:
As of draft-ietf-dnsop-dns-zone-digest-06 the protocol skips all apex ZONEMD RRs
Also skip RRSIG(ZONEMD) RRs at apex only

wessels committed:
Add SHA512 algorithm support

wessels committed:
Fix ZONEMD to_text() so that it correctly writes scheme and algorithm
Ignore schemes other than 1 (SIMPLE) when verifying

wessels committed:
original code wasn't handling multi-line or multi-word ZONEMD
digests.  Now mimicking what dnspython does for DS RRs.

wessels committed:
dnspython does this sort of horrible thing where it separates SIG/RRSIG
covering different types into different rdatasets.  Fortunately we can
work around this with a custom sorting function.

wessels committed:
remove stray debugging

shane-kerr commented on zonemd.py:
```diff
@@ -174,12 +189,17 @@ def add_zonemd(zone, zonemd_algorithm='sha384', zonemd_ttl=None):
+def rdataset_sorter(rdataset):
+    wire_rdata = rdataset[0].to_digestable()
+    wire_rdata_len = len(wire_rdata)
+    return (rdataset.rdtype, wire_rdata_len, wire_rdata)
```
Is this correct? Should this just be:
```python
    return (rdataset.rdtype, wire_rdata)
```

wessels replied:
Oh yes, I believe you are correct!

shane-kerr reviewed (commented):
Looks really good, thanks for the PR!

I'm not sure about the sorting though. Can you please have a look at the inline question and let me know (the version in the PR might be correct, I just want to double check).

wessels committed:
Oops, length of wire rdata doesn't play a role in sorting, except when
they differ in size but are identical up to the length of the smaller
one.  But in this case the normal binary comparison should suffice.
If not then it would be appropriate to add the length as the third
comparison field, not the second.
